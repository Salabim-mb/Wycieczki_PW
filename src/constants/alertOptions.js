const alertStyle = {
	ERROR: "error",
	WARNING: "warning",
	INFO: "info",
	SUCCESS: "success"
}

export default alertStyle;