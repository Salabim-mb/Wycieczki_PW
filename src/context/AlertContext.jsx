import React, {useState} from 'react';
import alertStyle from 'constants/alertOptions';

export const AlertContext = React.createContext({
	open: false,
	message: "",
	showAlert: () => {}
});

export const AlertProvider = (props) => {
	const [open, setOpen] = useState(false);
	const [message, setMessage] = useState("");
	const [severity, setSeverity] = useState(alertStyle.ERROR);

	const data = {
		open,
		message,
		severity,
		changeVisibility: (newOpen) => setOpen(newOpen),
		changeMessage: (newMessage, newSeverity=alertStyle.ERROR) => {
			setOpen(true);
			setMessage(newMessage);
			setSeverity(newSeverity);
		},
		showAlert: (newMessage, newSeverity) => {
			data.changeMessage(newMessage, newSeverity);
			data.changeVisibility(true);
		},
	};

	return <AlertContext.Provider value={data} {...props} />
};