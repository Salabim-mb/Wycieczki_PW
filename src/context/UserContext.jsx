import React, { useState } from 'react';
import Cookies from 'universal-cookie';
import { decryptCookie, encryptCookie } from 'utils/aesCiphrator';

const cookie = new Cookies();

export const UserContext = React.createContext({
	token: undefined,
	data: undefined,
	type: undefined,
	login: () => {},
	logout: () => {}
});

export const UserProvider = (props) => {
	const [token, setToken] = useState(
		decryptCookie(cookie.get("token")) || undefined
	);
	const [data, setData] = useState(
		decryptCookie(cookie.get("data")) || undefined
	);
	const [type, setType] = useState(
		decryptCookie(cookie.get("type")) || undefined
	)

	const user = {
		token,
		data,
		type,
		login: (newToken, newData, newType) => {
			setToken(newToken);
			setData(newData);
			setType(newType);
			cookie.set("token", encryptCookie(newToken), {
				path: "/",
				secure: process.env.REACT_APP_SECURE_COOKIES !== 'off'
			});
			cookie.set("data", encryptCookie(newData), {
				path: "/",
				secure: process.env.REACT_APP_SECURE_COOKIES !== 'off'
			});
			cookie.set("type", encryptCookie(newType), {
				path: "/",
				secure: process.env.REACT_APP_SECURE_COOKIES !== 'off'
			});
		},
		logout: () => {
			cookie.remove("token");
			cookie.remove("data");
			cookie.remove("type");
			setData(undefined);
			setToken(undefined);
			setType(undefined);
		},
	};

	return <UserContext.Provider value={user} {...props} />;
};