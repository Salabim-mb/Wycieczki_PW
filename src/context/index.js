export * from './AlertContext';
export * from './UserContext';
