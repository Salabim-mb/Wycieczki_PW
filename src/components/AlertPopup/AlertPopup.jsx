import React, { useContext, useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Snackbar from '@material-ui/core/Snackbar';
import { AlertContext } from 'context/AlertContext';
import MuiAlert from '@material-ui/lab/Alert'

const useStyles = makeStyles((theme) => ({
	root: {
		width: '100%',
		'& > * + *': {
			marginTop: theme.spacing(2),
		}
	}
}));

const Alert = (props) => (
	<MuiAlert elevation={6} variant="filled" {...props} />
);

const AlertPopup = () => {
	const alertC = useContext(AlertContext);
	const [snackPack, setSnackPack] = React.useState([]);
	const [open, setOpen] = React.useState(false);
	const [messageInfo, setMessageInfo] = React.useState(undefined);

	useEffect(() => {
		if (alertC.open) {
			setOpen(false)
			setSnackPack((prev) => [...prev, {
				message: alertC?.message,
				key: new Date().getTime(),
				severity: alertC.severity
			}]);
			setOpen(true);
		}
	}, [alertC.open, alertC.message])

	useEffect(() => {
		if (snackPack.length && !messageInfo) {
			setMessageInfo({ ...snackPack[0] });
			setSnackPack((prev) => prev.slice(1));
			setOpen(true);
		} else if (snackPack.length && messageInfo && open) {
			setOpen(false);
		}
	}, [snackPack, messageInfo, open]);

	const handleClose = (event, reason) => {
		if (reason === 'clickaway') {
			return;
		}
		setOpen(false);
		alertC.changeVisibility(false);
	};

	const handleExited = () => {
		setMessageInfo(undefined);
	};

	const classes = useStyles();
	return (
		<>
			<div key={messageInfo?.message} className={classes.root}>
				<Snackbar open={open} autoHideDuration={3000} onClose={handleClose} onExited={handleExited}>
					<Alert severity="error" onClose={handleClose}>
						{messageInfo?.message}
					</Alert>
				</Snackbar>
			</div>
		</>
	);
}

export default AlertPopup;