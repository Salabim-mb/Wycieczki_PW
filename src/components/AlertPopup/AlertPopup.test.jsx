import React from 'react';
import {render} from '@testing-library/react';
import alertOptions from 'constants/alertOptions';
import { AlertContext } from 'context/AlertContext';
import AlertPopup from './AlertPopup';

describe("AlertPopup", () => {
	let contextData;
	beforeEach(() => {
		contextData = {
			open: true,
			message: "test123",
			severity: alertOptions.ERROR,
			changeVisibility: jest.fn(),
			showAlert: jest.fn()
		}
		jest.clearAllMocks();
	})

	it("should match snapshot", () => {
		const {container} = render(
			<AlertContext.Provider value={contextData}>
				<AlertPopup />
			</AlertContext.Provider>
		);
		expect(container).toMatchSnapshot();
	});

	it("should display correct alert message", () => {
		const {getByText} = render(
			<AlertContext.Provider value={contextData}>
				<AlertPopup />
			</AlertContext.Provider>
		);
		expect(getByText(contextData.message)).toBeInTheDocument();
	});
});