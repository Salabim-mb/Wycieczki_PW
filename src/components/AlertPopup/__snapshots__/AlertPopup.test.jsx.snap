// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertPopup should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation6"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          test123
        </div>
        <div
          class="MuiAlert-action"
        >
          <button
            aria-label="Close"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            title="Close"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
